test/image-tests/aws.sh

```diff
@@ -0,0 +1,256 @@
+fi
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+# Apply lorax patch to work around pytoml issues in RHEL 8.x.
+# See BZ 1843704 or https://github.com/weldr/lorax/pull/1030 for more details.
+if [[ $ID == rhel ]]; then
+    sudo sed -r -i 's#toml.load\(args\[3\]\)#toml.load(open(args[3]))#' \
+        /usr/lib/python3.6/site-packages/composer/cli/compose.py
+    sudo rm -f /usr/lib/python3.6/site-packages/composer/cli/compose.pyc
+fi
+
+# We need jq for parsing composer-cli output.
+if ! hash jq; then
+    greenprint "Installing jq"
+    sudo dnf -qy install jq
+fi
+
+# Check for awscli.
+if ! hash aws; then
+    greenprint "Installing awscli"
+    sudo pip3 -qq install awscli
+fi
+
+TEST_UUID=$(uuidgen)
+IMAGE_KEY=osbuild-composer-aws-test-${TEST_UUID}
+AWS_CMD="aws --region $AWS_REGION --output json --color on"
+
+# Jenkins sets WORKSPACE to the job workspace, but if this script runs
+# outside of Jenkins, we can set up a temporary directory instead.
+if [[ ${WORKSPACE:-empty} == empty ]]; then
+    WORKSPACE=$(mktemp -d)
+fi
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+AWS_CONFIG=${TEMPDIR}/aws.toml
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+AWS_INSTANCE_JSON=${TEMPDIR}/aws-instance.json
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+AMI_DATA=${TEMPDIR}/ami-data-${IMAGE_KEY}.json
+INSTANCE_DATA=${TEMPDIR}/instance-data-${IMAGE_KEY}.json
+INSTANCE_CONSOLE=${TEMPDIR}/instance-console-${IMAGE_KEY}.json
+
+# Check for the smoke test file on the AWS instance that we start.
+smoke_test_check () {
+    # Ensure the ssh key has restricted permissions.
+    SSH_KEY=${WORKSPACE}/test/keyring/id_rsa
+    chmod 0600 $SSH_KEY
+
+    SMOKE_TEST=$(ssh -i ${SSH_KEY} redhat@${1} 'cat /etc/smoke-test.txt')
+    if [[ $SMOKE_TEST == smoke-test ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+# Get the console screenshot from the AWS instance.
+store_instance_screenshot () {
+    INSTANCE_ID=${1}
+    LOOP_COUNTER=${2}
+    SCREENSHOT_FILE=${WORKSPACE}/console-screenshot-${ID}-${VERSION_ID}-${LOOP_COUNTER}.jpg
+
+    $AWS_CMD ec2 get-console-screenshot --instance-id ${1} > $INSTANCE_CONSOLE
+    jq -r '.ImageData' | base64 -d - > $SCREENSHOT_FILE
+}
+
+# Write an AWS TOML file
+tee $AWS_CONFIG > /dev/null << EOF
+provider = "aws"
+
+[settings]
+accessKeyID = "${AWS_ACCESS_KEY_ID}"
+secretAccessKey = "${AWS_SECRET_ACCESS_KEY}"
+bucket = "${AWS_BUCKET}"
+region = "${AWS_REGION}"
+key = "${IMAGE_KEY}"
+EOF
+
+# Write a basic blueprint for our image.
+tee $BLUEPRINT_FILE > /dev/null << EOF
+name = "bash"
+description = "A base system with bash"
+version = "0.0.1"
+
+[[packages]]
+name = "bash"
+
+[customizations.services]
+enabled = ["sshd", "cloud-init", "cloud-init-local", "cloud-config", "cloud-final"]
+EOF
+
+# Copy the internal Fedora repository into place.
+if curl -fs http://download.devel.redhat.com > /dev/null; then
```

Comment:
Should this be specific to only this test?

Reply:
We could move it into a shared `function.sh` at some point when I add the next test for qcow. It could easily be used for other tests.